s3_storage_provider.py

```diff
@@ -84,6 +84,19 @@ def __init__(self, hs, config):
         if "session_token" in config:
             self.api_kwargs["aws_session_token"] = config["session_token"]
 
+        if "request_checksum_calculation" in config or "response_checksum_validation" in config:
+            from botocore.config import Config
```

Comment:
nit: Although this isn't mentioned in the code style guide, typically we only import something inline if otherwise doing so at the top of the file would cause a cyclical import. And in that case, we annotate why an inline import was needed in a comment.

So for consistencies sake, please move this import to the top of the file.